compiler/rustc_pattern_analysis/src/lints.rs

```diff
@@ -209,34 +209,19 @@ pub(crate) fn lint_nonexhaustive_missing_variants<'a, 'p, 'tcx>(
 
 /// Traverse the patterns to warn the user about ranges that overlap on their endpoints.
 #[instrument(level = "debug", skip(cx))]
-pub(crate) fn lint_overlapping_range_endpoints<'a, 'p, 'tcx>(
+pub(crate) fn collect_overlapping_range_endpoints<'a, 'p, 'tcx>(
     cx: MatchCtxt<'a, 'p, 'tcx>,
     column: &PatternColumn<'p, 'tcx>,
+    overlapping_range_endpoints: &mut Vec<rustc::OverlappingRanges<'p, 'tcx>>,
```

Comment:
Performance is irrelevant for this so it's a stylistic choice. I personally prefer to be computing data than calling callbacks in terms of code legibility. Otherwise I'd have made that a method on TypeCx I guess

Reply:
I think it'll depend a lot of what's possible on the rust-analyzer side as well. If it fits in TypeCx on their side too I might change to that

Reply:
I don't know what I was thinking, you're quite right actually